dotnet-authserver/src/TeacherIdentity.AuthServer/Pages/SignIn/Register/PhoneConfirmation.cshtml.cs

```diff
@@ -43,7 +49,25 @@ public async Task<IActionResult> OnPost()
             return await HandlePinVerificationFailed(pinVerificationFailedReasons);
         }
 
-        HttpContext.GetAuthenticationState().OnMobileNumberVerified();
+        var authenticationState = HttpContext.GetAuthenticationState();
+        var permittedUserTypes = authenticationState.GetPermittedUserTypes();
+
+        var user = await _dbContext.Users
+            .Where(u => (u.MobileNumber ?? "").EndsWith(NormaliseMobileNumber(MobileNumber!)))
```

Comment:
Unfortunately we have to do some formatting of the mobile number before storing or this lookup will be inefficient. At the moment formatting should be done when the mobile number is saved to the user model (not yet implemented), but can perhaps be done earlier in e.g. register/phone

Reply:
I think as you have it here is best; ideally the value in `AuthenticationState` is as the user entered it, at least for the registration process.